Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeAlignmentAlgorithm.cc

```diff
@@ -582,6 +585,17 @@ MillePedeAlignmentAlgorithm::addHitCount(const std::vector<AlignmentParameters*>
   return nHitY;
 }
 
+
+void MillePedeAlignmentAlgorithm::beginRun(const edm::Run& run,
+                                           const edm::EventSetup& setup) {
+  if (run.run() < firstIOV_ && !ignoreFirstIOVCheck_) {
+    throw cms::Exception("Alignment")
+      << "@SUB=MillePedeAlignmentAlgorithm::beginRun\n"
+      << "Using data (run = " << run.run()
+      << ") prior to the first defined IOV (" << firstIOV_ << ").";
+  }
```

Comment:
Hi @franzoni,
yes, I have seen this failure, but this check has a different purpose.
This is checking the output IOV definition, e.g. if the alignment is supposed to deliver IOVs starting from run 200000 and data is used from , e.g., run 190000.
Without the check, the resulting alignment of the high-level structures would not be affected because the data does not fall into any IOV, but the module-level alignment would be slightly affected because the modules are assumed to be constant wrt. the high-level structures, i.e. all data is used for the module-level alignment. That means data from all IOVs and without this check even from periods prior to any defined output IOV.
In addition this check also prevents problems in a second iteration where the input IOVs are the output IOVs from the first iteration. In this case, we would get the kind of message you are pointing to, but only after running an alignment which might have taken up to two days.